massa-protocol-worker-2/src/handlers/block_handler/propagation.rs

```diff
@@ -39,18 +43,28 @@ impl PropagationThread {
                                 { "block_id": block_id }
                             );
                             let header = {
-                                let blocks = storage.read_blocks();
-                                blocks
-                                    .get(&block_id)
-                                    .map(|block| block.content.header.clone())
-                                    .ok_or_else(|| {
-                                        ProtocolError::ContainerInconsistencyError(format!(
-                                            "header of id {} not found.",
-                                            block_id
-                                        ))
-                                    })
-                                    .unwrap()
+                                let block = {
+                                    let blocks = storage.read_blocks();
+                                    blocks.get(&block_id).cloned()
+                                };
+                                if let Some(block) = block {
+                                    self.storage.store_block(block.clone());
+                                    self.saved_blocks.push_back(block.id);
+                                    if self.saved_blocks.len()
```

Comment:
maybe check len() + 1 before insert on it and not after + rollback ?

Reply:
I don't rollback I remove the oldest one that is at front but keep the one I just inserted which is at the back.

Reply:
oh yeah sorry !